index.ts

```diff
@@ -258,6 +263,69 @@ export class Project {
       dep.baseDir = path.join(this.baseDir, 'node_modules', dep.name);
       dep.writeSync();
     }
+    for (let [name, { dir: target }] of this.dependencyLinks) {
+      this.writeLinkedPackage(name, target);
+    }
+  }
+
+  private writeLinkedPackage(name: string, target: string) {
+    let targetPkg = fs.readJsonSync(`${target}/package.json`);
+    let peers = new Set(Object.keys(targetPkg.peerDependencies ?? {}));
+    let destination = path.join(this.baseDir, 'node_modules', name);
+
+    if (peers.size === 0) {
+      // no peerDeps, so we can just symlink the whole package
+      fs.ensureSymlinkSync(target, destination, 'dir');
+      return;
+    }
+
+    // need to reproduce the package structure in our own location
+    this.hardLinkContents(target, destination);
+
+    for (let section of ['dependencies', 'peerDependencies']) {
+      if (targetPkg[section]) {
+        for (let depName of Object.keys(targetPkg[section])) {
+          if (peers.has(depName)) {
+            continue;
+          }
+          let depTarget = resolvePackagePath(depName, target, this.resolutionCache);
+          if (!depTarget) {
+            throw new Error(
+              `[FixturifyProject] package ${name} in ${target} depends on ${depName} but we could not resolve it`
+            );
+          }
+          fs.ensureSymlinkSync(
+            depTarget.slice(0, -1 * '/package.json'.length),
```

Comment:
will this work correctly if `depTarget` contains windows path separator? Could `path.dirname(depTarget)` work without caveats?